File: android-client/app/src/main/java/com/omnicart/agent/core/model/DecisionResult.kt
package com.omnicart.agent.core.model

import com.google.gson.annotations.SerializedName

data class DecisionResult(
    @SerializedName("product_id")
    val productId: String = "",
    @SerializedName("final_score")
    val finalScore: Double = 0.0,
    @SerializedName("display_score")
    val displayScore: Double = 0.0,
    @SerializedName("score_breakdown")
    val scoreBreakdown: ScoreBreakdown? = null,
    @SerializedName("component_scores")
    val componentScores: Map<String, Map<String, Any?>>? = null,
    @SerializedName("evidence_ids")
    val evidenceIds: List<String> = emptyList(),
    @SerializedName("risk_factors")
    val riskFactors: List<String> = emptyList(),
    @SerializedName("recommendation_reason")
    val recommendationReason: String = "",
    @SerializedName("llm_relevance")
    val llmRelevance: Double = 0.0,
    @SerializedName("llm_reasoning")
    val llmReasoning: String = "",
    @SerializedName("llm_verdict")
    val llmVerdict: String = "",
    @SerializedName("recommendation_level")
    val recommendationLevel: String = "",
    @SerializedName("evidence_confidence")
    val evidenceConfidence: Double = 0.0,
    @SerializedName("support_evidence_ids")
    val supportEvidenceIds: List<String> = emptyList(),
)

/**
 * V4 ScoreBreakdown — 旧 7 维命名（保留向后兼容）。
 * 注意: visual_similarity 实际存的是 relevance (RAG相关度), availability_score 实际存的是 value_score (性价比)。
 * 新代码优先使用 component_scores (9维结构化输出)。
 */
data class ScoreBreakdown(
    @SerializedName("budget_fit")
    val budgetFit: Double = 0.0,
    @SerializedName("scenario_fit")
    val scenarioFit: Double = 0.0,
    @SerializedName("spec_match")
    val specMatch: Double = 0.0,
    @SerializedName("review_confidence")
    val reviewConfidence: Double = 0.0,
    @SerializedName("visual_similarity")   // 实际: relevance (RAG相关度)
    val visualSimilarity: Double = 0.0,
    @SerializedName("availability_score")   // 实际: value_score (性价比)
    val availabilityScore: Double = 0.0,
    @SerializedName("risk_penalty")
    val riskPenalty: Double = 0.0
)

File: app/schemas/decision_result.py
"""
“最终判决文书”！
它出现在整个链路的最后一道关卡（decision_node），
决定了哪 3 件商品（D5 固定 Top-3）能真正展示给用户。
7 维分项 + 证据绑定 + 推荐等级

"""
from typing import Optional
from pydantic import BaseModel, Field

#极其丰富的 7+ 维度打分体系 (ScoreBreakdown)
class ScoreBreakdown(BaseModel):
    budget_fit: float = 0.0
    scenario_fit: float = 0.0
    spec_match: float = 0.0
    review_confidence: float = 0.0
    visual_similarity: float = 0.0
    availability_score: float = 1.0
    risk_penalty: float = 0.0
    # P2: Memory-aware dimensions
    preference_match_score: float = 0.0
    device_compatibility_score: float = 0.0
    brand_preference_boost: float = 0.0
    avoid_tag_penalty: float = 0.0

# LLM 评估与公式打分的完美融合
class DecisionResult(BaseModel):
    product_id: str
    final_score: float = 0.0
    display_score: float = 0.0
    score_breakdown: ScoreBreakdown = Field(default_factory=ScoreBreakdown)
    evidence_ids: list[str] = Field(default_factory=list)
    risk_factors: list[str] = Field(default_factory=list)
    recommendation_reason: str = ""
    # P2: Memory trace reference
    memory_contributions: list[dict] = Field(default_factory=list)
    # V2: LLM Evidence Evaluation
    llm_relevance: float = 0.0
    llm_reasoning: str = ""
    llm_verdict: str = ""  # strong_recommend | recommend | consider | avoid
    # V4: Evidence-Grounded Scoring
    score_version: str = "evidence_scoring_v1"
    suitability_score: Optional[float] = None
    evidence_confidence: Optional[float] = None
    component_scores: dict = Field(default_factory=dict)
    support_evidence_ids: list[str] = Field(default_factory=list)
    recommendation_level: str = ""
    hard_constraint_status: str = "pass"
    scoring_debug: dict = Field(default_factory=dict)

